packages/flutter_tools/templates/plugin/android-java.tmpl/src/main/java/androidIdentifier/pluginClass.java.tmpl

```diff
@@ -1,5 +1,31 @@
 package {{androidIdentifier}};
 
+{{#useNewAndroidEmbedding}}
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+/** {{pluginClass}} */
+public class {{pluginClass}} implements FlutterPlugin, MethodCallHandler {
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding flutterPluginBinding) {
+    final MethodChannel channel = new MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "{{projectName}}");
+    channel.setMethodCallHandler(new {{pluginClass}}());
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
```

Comment:
Can we generate 2 files instead of 1? I think we should incentivize channel comms in a dedicated file so that people might actually unit test message serialization.

Reply:
The templates in this directory is used for plugin projects that don't use the new embedding. However, if 1 file is applicable to the new embedding only, then we would need to duplicate the directory or refactor the code to exclude certain files. I can follow up with this later, wdyt?

Reply:
The top of this template says "implements FlutterPlugin", which is the new embedding.

Reply:
This file generates code for both embedding. see `{{#useNewAndroidEmbedding}}` and `{{^useNewAndroidEmbedding}}` sections.